main.py

```diff
@@ -220,7 +255,7 @@ def run_inventory_loop(config, connection):
         configuration['servers'] = call_args.servers
 
     netbox_connection = NetboxConnection(configuration)
-
+    inventory_obj.runSerialNumberScript(call_args.query)
```

Comment:
Ì don't think this is the right place to call the function.

In the lines below depending on the arguments we are either starting a falcon webserver to receive calls to query a device or we just run in a loop over a list of servers either given in the config or from a netbox query.

It looks to me that your class already needs a device as an argument during object creation. But at this time we have no device yet to query. We have handler.py for that.

Did you test the code locally? Did it work like that for you?